File: dotnet/Allmark/Render/ConsoleHtmlRenderer.cs
namespace Allmark.Render;

using Allmark.Types;

public static class ConsoleHtmlRenderer
{
    public static Renderer Create()
    {
        return new Renderer
        {
            Name = "html",
            Render = Render,
        };
    }

    public static void Render(MarkdownNode node, RendererState state, bool? first = null, bool? last = null, bool? decode = true)
    {
        state.Output.Append(node.Content);
    }
}

File: zig/src/render-console/consoleHtmlRenderer.zig
const std = @import("std");

const MarkdownNode = @import("../types/MarkdownNode.zig").MarkdownNode;
const ConsoleRendererState = @import("../types/RendererState.zig").RendererState;
const Renderer = @import("../types/Renderer.zig").Renderer;

pub const consoleHtmlRenderer = Renderer{
    .name = "html",
    .render = render,
};

fn render(node: *const MarkdownNode, state: *ConsoleRendererState, decode: ?bool) void {
    _ = decode;

    state.output.appendSlice(state.allocator, node.content) catch unreachable;
}
